src/Features/CSharp/Portable/PullMemberUp/CSharpMembersPullerService.cs

```diff
@@ -27,60 +26,31 @@ public CSharpMembersPullerService()
         {
         }
 
-        protected override SyntaxTriviaList GetFirstMemberTrivia(SyntaxNode root)
+        protected override SyntaxList<SyntaxNode> GetMembers(SyntaxNode root)
         {
-            var members = root switch
+            return root switch
             {
                 CompilationUnitSyntax compilationUnit => compilationUnit.Members,
                 NamespaceDeclarationSyntax namespaceDeclaration => namespaceDeclaration.Members,
                 _ => throw ExceptionUtilities.UnexpectedValue(root)
             };
-
-            if (members.Count == 0)
-            {
-                return SyntaxTriviaList.Empty;
-            }
-
-            return members.First().GetLeadingTrivia();
         }
 
-        protected override SyntaxNode EnsureLeadingTriviaBeforeFirstMember(SyntaxNode root, SyntaxTriviaList trivia)
-        {
-            var members = root switch
-            {
-                CompilationUnitSyntax compilationUnit => compilationUnit.Members,
-                NamespaceDeclarationSyntax namespaceDeclaration => namespaceDeclaration.Members,
-                _ => throw ExceptionUtilities.UnexpectedValue(root)
-            };
-
-            if (members.Count == 0)
-            {
-                return root;
-            }
-
-            var firstMember = members.First();
-            return root.ReplaceNode(firstMember, firstMember.WithLeadingTrivia(trivia));
-        }
-
-        protected override async Task<IEnumerable<UsingDirectiveSyntax>> GetImportsAsync(Document document, CancellationToken cancellationToken)
+        protected override async Task<ImmutableArray<UsingDirectiveSyntax>> GetImportsAsync(Document document, CancellationToken cancellationToken)
         {
             var root = await document.GetRequiredSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
             return root.DescendantNodesAndSelf()
-                .Where((node) => node is CompilationUnitSyntax || node is NamespaceDeclarationSyntax)
+                .Where(node => node is CompilationUnitSyntax || node is NamespaceDeclarationSyntax)
                 .SelectMany(node => node switch
                 {
                     CompilationUnitSyntax c => c.Usings,
                     NamespaceDeclarationSyntax n => n.Usings,
-                    _ => default,
+                    _ => throw ExceptionUtilities.UnexpectedValue(node),
                 })
-                .Distinct()
-                .Select(import => import.NormalizeWhitespace().WithTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed));
+                .SelectAsArray(import => import.NormalizeWhitespace().WithTrailingTrivia(SyntaxFactory.ElasticCarriageReturnLineFeed));
```

Comment:
so this seems to be going in the opposite directino of what i woudl expect.  instead starting at the root and walking downwards, we shoudl start on the type we're pulling members from, and walk upwards to see what imports are in scope.

When you walk downwards you may hit lots of irrelevent imports (like those in *different* namespace blocks).  Walking upwards hits only the relevant ones.

Note: this will cause us to find all imports, but there's a case this still misses.  Imagine we have:

```c#
namespace N1
{
    class Base { }
}

namespace N2
{
    class Derived : Base { ... members ... }
}
```

If we pull up members from Base to Derived, then anything that bound to a symbol in N2 will not bind once inside N1.  We can either leave this as a whole, or we can attempt to add a `using N2` into the destination, which we can remove if not used.

Reply:
also, why are we normalizing the whitespace?